### ISSUE TITLE
Federation documentation suggest invalid configuration

### ISSUE THREAD

**Issue by DaDummy** [RocketChat/docs-old#2001]
The Federation documentation currently suggests creating SRV and TXT records on the subdomain the rocket chat instance is hosted on (`chat.mydomain.com`) rather than the domain that is assumed to be the federation domain (`mydomain.com`) and that is also seen in the domain field in the configuration screenshot:

```
dig srv _rocketchat._https.chat.mydomain.com
...
;; ANSWER SECTION:
_rocketchat._https.chat.mydomain.com. 1800 IN SRV    1 1 443 chat.mydomain.com.
...
​
dig -t txt rocketchat-public-key.chat.mydomain.com
...
;; ANSWER SECTION:
rocketchat-public-key.chat.mydomain.com. 1799 IN TXT "-----BEGIN PUBLIC KEY----- THEPUBLICKEY :) -----END PUBLIC KEY-----"
...
```

This however would result in an invalid configuration and the "Test setup" button will keep complaining.

The correct setup instead looks like this:
```
dig srv _rocketchat._https.mydomain.com
...
;; ANSWER SECTION:
_rocketchat._https.mydomain.com. 1800 IN SRV    1 1 443 chat.mydomain.com.
...
​
dig txt rocketchat-public-key.mydomain.com
...
;; ANSWER SECTION:
rocketchat-public-key.mydomain.com. 1799 IN TXT "-----BEGIN PUBLIC KEY----- THEPUBLICKEY :) -----END PUBLIC KEY-----"
...
```

**Comment by soham4abc** [comment]
I am working on this issue!

**Comment by DaDummy** [comment]
While the merged change fixes the reported issue, it also changes the portrayed scenario to a simpler, less helpful one :/

It would have been a better improvement to keep the assumption of having the server hosted at `chat.mydomain.com` as that is a probably far more common scenario. But well, in the end that is your decision not mine.

**Comment by ChristySujitha** [comment]
Resolved the issue (federation documentation invalid configuration resolved #2002). Here is the updated the doc: https://docs.rocket.chat/guides/administration/admin-panel/settings/federation#recommended-configure-your-dns.
